src/internal/kopia/wrapper.go

```diff
@@ -399,14 +399,23 @@ func loadDirsAndItems(
 			return nil, el.Failure()
 		}
 
-		ictx := clues.Add(ctx, "restore_path", col.restorePath)
+		var (
+			mergeDC *mergeCollection
+			ictx    = clues.Add(ctx, "restore_path", col.restorePath)
+			merge   = len(col.storageDirs) > 0
+		)
+
+		if merge {
+			mergeDC = &mergeCollection{fullPath: col.restorePath}
+			res = append(res, mergeDC)
+		}
```

Comment:
What if, instead of the conditional, you made everything a mergeCollection?  That would simplify the code a bit and remove some conditional paths (plus extra mental load about state).  And any mergeCollection with a single item should still behave as expected.